src/seedu/addressbook/data/person/Address.java

```diff
@@ -22,13 +32,28 @@
      */
     public Address(String address, boolean isPrivate) throws IllegalValueException {
         this.isPrivate = isPrivate;
+        System.out.println(address);
         if (!isValidAddress(address)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
-        this.value = address;
+        
+        value = address;
+        parseAndAssignAddress(address);
     }
 
-    /**
+    private void parseAndAssignAddress(String address) {
+		String[] splittedAddress = address.split(ADDRESS_SPLITTER);
+		block = parseBlock(splittedAddress[BLOCK_INDEX]);
+		street = new Street(splittedAddress[STREET_INDEX]);
+		unit = new Unit(splittedAddress[UNIT_INDEX]);
+		postalCode = new PostalCode(Integer.valueOf(splittedAddress[POSTAL_CODE_INDEX]));
+	}
+    
+    Block parseBlock(String block) {
+    	return new Block(block.substring(BLOCK_SUBSTRING_START, block.length() - 1));
+    }
```

Comment:
Too many things going on here. Use more abstraction. The reader might wonder why parsing results in creating a `Block` object.